src/main/java/com/example/movieapp/controllers/MovieController.java

```diff
@@ -0,0 +1,42 @@
+package com.example.movieapp.controllers;
+
+import com.example.movieapp.dto.MovieDTO;
+import com.example.movieapp.entities.Movie;
+import com.example.movieapp.entities.Producer;
+import com.example.movieapp.exceptions.ParentRecordNotFoundException;
+import com.example.movieapp.repositories.MovieRepository;
+import com.example.movieapp.repositories.ProducerRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+import static com.example.movieapp.enums.ErrorCode.PRODUCER_NOT_FOUND;
+
+@RestController
+@RequestMapping("/movies")
+public class MovieController {
+
+    @Autowired
+    MovieRepository movieRepository;
+
+    @Autowired
+    ProducerRepository producerRepository;
+
+    @GetMapping("")
+    public List<Movie> getAllProducers() {
+        return movieRepository.findAll();
+    }
+
+    @PostMapping("")
+    public ResponseEntity<Movie> createMovie(@RequestBody MovieDTO movieDTO) {
```

Comment:
I think it is worth to move all the logic (exception handling, mapping, save operations) to service layer, controller layer should be quite thin